src/SearchQueryNormalizer.js

```diff
@@ -0,0 +1,149 @@
+
+const ModelWrapper = require('./ModelWrapper');
+const error = require('./error');
+
+const defaultSupportedOpperators = [
+    '=',
+    'neq',
+    'lt',
+    'lte',
+    'gt',
+    'gte',
+    'like',
+    'nlike',
+    'ilike',
+    'nilike',
+    'inq',
+    'nin',
+    'between',
+];
+
+/**
+ * @todo: add a custom error type for easier handling
+ * @todo: add a possibility to detect invalid operators
+ */
+
+module.exports = class SearchQueryNormalizer {
+
+    constructor(models, {
+        supportedOperators = defaultSupportedOpperators,
+        rejectUnknownProperties = false,
+    } = {}) {
+        this.models = models;
+        this.wrappedModels = {};
+        this.supportedOperators = supportedOperators;
+        this.rejectUnknownProperties = rejectUnknownProperties;
+    }
+
+    normalizeQuery(rootModelName, where) {
+        return this.normalizeWhereQuery(rootModelName, where);
+    }
+
+    setUnknownPropertyRejection(rejectUnknown) {
+        this.rejectUnknownProperties = rejectUnknown === true;
+    }
+
+    normalizeWhereQuery(rootModelName, where) {
```

Comment:
a general summary of what your functions / methods do would be nice if the function is a bit more than a onliner